src/ui/ui_grid.cpp

```diff
@@ -65,7 +66,33 @@ UIGrid::UIGrid(float start_x, float start_y,
     // Set up callbacks.
     // TODO: add semantic game logic abstraction layer
     events::joystick_event.connect([&](events::JoystickData d) {
-            move_selection(Direction::DOWN);});
+        if (Phase::curr_phase != PhaseType::BUILD) 
+            return;
+        if (d.is_button == true && d.input == 1 && d.state == 0)
+            events::build::construct_changed_event(ConstructType::NONE);
+        if (!BuildPhase::is_menu) 
```

Comment:
As you hinted, I think a game semantic translation layer is warranted. This way, you don't have to check the ``Phase::curr_phase`` or some boolean of a specific Phase. Neither do you have to have any of the low level joystick input ``switch``ing at this UI Grid level. I am currently thinking of two options:
- Some ``ControlMapping`` or ``InputMapping`` or ``InputHandler`` module (this can in fact replace the dirty ``debug_input`` we have) that contains several mappings from raw input events to game events and switches between them as we switch between phases.
- Or the ``GameState`` has some handle_input and delegates to its ``curr_phase``, so the proper input handling is done without needing to externally check what phase we're in. (again, looking at how [he](http://gameprogrammingpatterns.com/state.html) does it)